sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala

```diff
@@ -118,309 +142,286 @@ class SQLListenerSuite extends SparkFunSuite with SharedSQLContext with JsonTest
-    assert(spark.sharedState.listener.stageIdToStageMetrics.size == previousStageNumber)
+    assert(statusStore.executionsList().size == previousStageNumber)
 
     spark.sparkContext.parallelize(1 to 10).toDF().foreach(i => ())
     spark.sparkContext.listenerBus.waitUntilEmpty(10000)
     // listener should save the SQL stage
-    assert(spark.sharedState.listener.stageIdToStageMetrics.size == previousStageNumber + 1)
-  }
-
-  test("SPARK-13055: history listener only tracks SQL metrics") {
-    val listener = new SQLHistoryListener(sparkContext.conf, mock(classOf[SparkUI]))
-    // We need to post other events for the listener to track our accumulators.
-    // These are largely just boilerplate unrelated to what we're trying to test.
-    val df = createTestDataFrame
-    val executionStart = SparkListenerSQLExecutionStart(
-      0, "", "", "", SparkPlanInfo.fromSparkPlan(df.queryExecution.executedPlan), 0)
-    val stageInfo = createStageInfo(0, 0)
-    val jobStart = SparkListenerJobStart(0, 0, Seq(stageInfo), createProperties(0))
-    val stageSubmitted = SparkListenerStageSubmitted(stageInfo)
-    // This task has both accumulators that are SQL metrics and accumulators that are not.
-    // The listener should only track the ones that are actually SQL metrics.
-    val sqlMetric = SQLMetrics.createMetric(sparkContext, "beach umbrella")
-    val nonSqlMetric = sparkContext.longAccumulator("baseball")
-    val sqlMetricInfo = sqlMetric.toInfo(Some(sqlMetric.value), None)
-    val nonSqlMetricInfo = nonSqlMetric.toInfo(Some(nonSqlMetric.value), None)
-    val taskInfo = createTaskInfo(0, 0)
-    taskInfo.setAccumulables(List(sqlMetricInfo, nonSqlMetricInfo))
-    val taskEnd = SparkListenerTaskEnd(0, 0, "just-a-task", null, taskInfo, null)
-    listener.onOtherEvent(executionStart)
-    listener.onJobStart(jobStart)
-    listener.onStageSubmitted(stageSubmitted)
-    // Before SPARK-13055, this throws ClassCastException because the history listener would
-    // assume that the accumulator value is of type Long, but this may not be true for
-    // accumulators that are not SQL metrics.
-    listener.onTaskEnd(taskEnd)
-    val trackedAccums = listener.stageIdToStageMetrics.values.flatMap { stageMetrics =>
-      stageMetrics.taskIdToMetricUpdates.values.flatMap(_.accumulatorUpdates)
-    }
-    // Listener tracks only SQL metrics, not other accumulators
-    assert(trackedAccums.size === 1)
-    assert(trackedAccums.head === ((sqlMetricInfo.id, sqlMetricInfo.update.get)))
+    assert(statusStore.executionsList().size == previousStageNumber + 1)
   }
 
   test("driver side SQL metrics") {
-    val listener = new SQLListener(spark.sparkContext.conf)
-    val expectedAccumValue = 12345
+    val oldCount = statusStore.executionsList().size
+    val expectedAccumValue = 12345L
     val physicalPlan = MyPlan(sqlContext.sparkContext, expectedAccumValue)
-    sqlContext.sparkContext.addSparkListener(listener)
```

Comment:
Does it really work? The listener is attached to a `ReplayBus`, not the real spark event bus.

Reply:
Yes, it works. There are two kinds of tests in this suite now:

- "test(blah)" like this one which uses the active spark session's listener
- "sqlStoreTest" which manually drive a replay bus and verify expected changes in the store.

This particular test (and the other one you commented on) are the first kind. Tests of the other kind do not run actual jobs, they just inject events into the replay bus manually.